File: TFScode/SiteFuel.Exchange/SiteFuel.FreightService/SiteFuel.MdbDataAccess/Collections/ChildObjects/DriverSchedule.cs
using System;
using System.Collections.Generic;
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;

namespace SiteFuel.MdbDataAccess.Collections
{
   public  class DriverSchedule
    {
       
        public String Id { get; set; }
        public ObjectId ShiftId { get; set; }
        public DateTimeOffset StartDate { get; set; }
        public DateTimeOffset EndDate { get; set; }
        public List<DateTimeOffset> RepeatDayList { get; set; }           
        public bool IsActive { get; set; }
        public string Description { get; set; }
        public int? RepeatEveryDay { get; set; }
        public int TypeId { get; set; }
    }
}

File: TFScode/SiteFuel.Exchange/SiteFuel.Exchange.SourceCode/SiteFuel.Exchange.Web/src/app/driver/models/driverSchedule.ts
export class DriverScheduleMapping {
    Id?: string;
    DriverId?: number;    
    DriverName?: string;    
    ScheduleList?: DriverSchedule[]=[];
}

export class  DriverSchedule {

    Id?: string;
    ShiftId? : string;
    StartDate?: Date;
    EndDate?: Date;
    RepeatDayList?: any[] = [];
    RepeatDayStringList?: any[] = [];
    Type?: number;
    IsActive?: boolean;
    Repeat?: number;  //daily or custom
    selectedShifts?: any[] = [];
    selectedRepeatList?: any[] = [];
    TypeId?:number;
    RepeatEveryDay?: number; 
    }

    export class ConflictDates {      
        Date: string;        
      }